src/ImageSharp/Dithering/ErrorDiffusion/IErrorDiffuser.cs

```diff
@@ -0,0 +1,34 @@
+﻿// <copyright file="IErrorDiffuser.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Dithering
+{
+    using System;
+
+    /// <summary>
+    /// Encapsulates properties and methods required to perfom diffused error dithering on an image.
+    /// </summary>
+    public interface IErrorDiffuser
+    {
+        /// <summary>
+        /// Gets the dithering matrix
+        /// </summary>
+        byte[][] Matrix { get; }
```

Comment:
Some `Fast2DArray<T>` benchmarks. Faster than a jagged array for most situations (including ours) and between 50-60% faster than 2D array across the board. Safer than jagged array also API -wise.

`Fast2DArray<T>`  FTW! Great suggestion @antonfirsov After I merge this PR i'll apply it to our convolution filters also.

```
BenchmarkDotNet=v0.10.1, OS=Microsoft Windows NT 6.2.9200.0
Processor=Intel(R) Core(TM) i7-6600U CPU 2.60GHz, ProcessorCount=4
Frequency=2742188 Hz, Resolution=364.6723 ns, Timer=TSC
  [Host]     : Clr 4.0.30319.42000, 64bit RyuJIT-v4.6.1586.0
  DefaultJob : Clr 4.0.30319.42000, 64bit RyuJIT-v4.6.1586.0

Allocated=0 B
```

Method | Count |      Mean |    StdErr |    StdDev | Scaled | Scaled-StdDev |
------------------------------------ |------ |---------- |---------- |---------- |------- |-------------- |
'Array access using 2D array' |    10 | 1.3680 ns | 0.0110 ns | 0.0395 ns |   1.00 |          0.00 |
'Array access using a jagged array' |    10 | 0.5169 ns | 0.0121 ns | 0.0453 ns |   0.38 |          0.03 |
'Array access using Fast2DArray' |    10 | 0.5076 ns | 0.0332 ns | 0.1243 ns |   0.37 |          0.09 |
'Array access using 2D array' |   100 | 1.0366 ns | 0.0370 ns | 0.1432 ns |   1.00 |          0.00 |
'Array access using a jagged array' |   100 | 0.4906 ns | 0.0101 ns | 0.0365 ns |   0.48 |          0.08 |
'Array access using Fast2DArray' |   100 | 0.5683 ns | 0.0240 ns | 0.0931 ns |   0.56 |          0.12 |
'Array access using 2D array' |  1000 | 0.9641 ns | 0.0405 ns | 0.1569 ns |   1.00 |          0.00 |
'Array access using a jagged array' |  1000 | 0.6315 ns | 0.0350 ns | 0.1308 ns |   0.67 |          0.16 |
'Array access using Fast2DArray' |  1000 | 0.5521 ns | 0.0307 ns | 0.1147 ns |   0.58 |          0.14 |
'Array access using 2D array' | 10000 | 1.0596 ns | 0.0380 ns | 0.1470 ns |   1.00 |          0.00 |
'Array access using a jagged array' | 10000 | 0.5105 ns | 0.0173 ns | 0.0624 ns |   0.49 |          0.08 |
'Array access using Fast2DArray' | 10000 | 0.4680 ns | 0.0211 ns | 0.0819 ns |   0.45 |          0.09 |